lib/network.go

```diff
@@ -3386,6 +3395,88 @@ func (txnData *LikeMetadata) New() DeSoTxnMetadata {
 	return &LikeMetadata{}
 }
 
+// ==================================================================
+// ReactMetadata
+//
+// A reaction is an interaction where a user on the platform reacts to a post.
+// ==================================================================
+
+type ReactMetadata struct {
+	// The user reacting is assumed to be the originator of the
+	// top-level transaction.
+
+	// The post hash to react to.
+	PostHash *BlockHash
+
+	// Set to true when a user is requesting to "remove" a reaction.
+	IsRemove bool
+
+	// The Unicode for the emoji reaction.
+	EmojiReaction rune
+}
+
+func (txnData *ReactMetadata) GetTxnType() TxnType {
+	return TxnTypeReact
+}
+
+func (txnData *ReactMetadata) ToBytes(preSignature bool) ([]byte, error) {
+	// Validate the metadata before encoding it.
+	//
+
+	var data []byte
+
+	// Add PostHash
+	//
+	// We know the post hash is set and has the expected length, so we don't need
+	// to encode the length here.
+	data = append(data, txnData.PostHash[:]...)
+
+	// Add IsRemove
+	data = append(data, BoolToByte(txnData.IsRemove))
+
+	// Add EmojiReaction.
+	// It is possible for a single character to be encoded with different code point sequences.
+	// By normalizing the Unicode (NFC), we ensure that a character will have a unique code point sequence.
+	data = append(data, norm.NFC.Bytes([]byte(string(txnData.EmojiReaction)))...)
```

Comment:
I would flip these two. Also, will the EmojiReaction always be the same exact length of bytes? if not, we will want to prepend the number of bytes.